src/tunnelbridge_map.h

```diff
@@ -42,6 +42,19 @@ static inline TransportType GetTunnelBridgeTransportType(TileIndex t)
 	return (TransportType)GB(_m[t].m5, 2, 2);
 }
 
+/**
+ * Tunnel: Is this tunnel a road tunnel?
+ * Bridge: Is the bridge's ramp a road?
+ * @param t The tile to analyze
+ * @pre IsTileType(t, MP_TUNNELBRIDGE)
+ * @return true if and only if the tunnel/bridge is a road tunnel/bridge
+*/
+static inline bool IsRoadTunnelBridge(TileIndex t)
```

Comment:
I feel less confident messing about in here but it made the code cleaner than replicating this check in `newgrf_roadtype.cpp`, even though IsRoadTunnelBridge is currently only used there.

Should we assert in this method? I'm guessing that helps better with stack traces but not certain given the same check is made in `GetTunnelBridgeTransportType()`

Reply:
I think assert is not needed here. But you need to also add `IsRoadTunnelBridgeTile()` (you can check how other XXXTile functions are done.